src/field_theory/abel_ruffini.lean

```diff
@@ -0,0 +1,373 @@
+  obtain ⟨c, hc⟩ := key σ,
+  obtain ⟨d, hd⟩ := key τ,
+  rw [σ.mul_apply, τ.mul_apply, hc, τ.map_mul, τ.commutes, hd, σ.map_mul, σ.commutes, hc],
+  rw [mul_assoc, mul_assoc, mul_right_inj' hb', mul_comm],
+end
+
+lemma splits_X_pow_sub_one_of_X_pow_sub_C {F : Type*} [field F] {E : Type*} [field E]
+  (i : F →+* E) (n : ℕ) {a : F} (ha : a ≠ 0) (h : (X ^ n - C a).splits i) : (X ^ n - 1).splits i :=
+begin
+  have ha' : i a ≠ 0 := mt (i.injective_iff.mp (i.injective) a) ha,
+  by_cases hn : n = 0,
+  { rw [hn, pow_zero, sub_self],
+    exact splits_zero i },
+  have hn' : 0 < n := pos_iff_ne_zero.mpr hn,
+  have hn'' : (X ^ n - C a).degree ≠ 0 :=
+    ne_of_eq_of_ne (degree_X_pow_sub_C hn' a) (mt with_bot.coe_eq_coe.mp hn),
+  obtain ⟨b, hb⟩ := exists_root_of_splits i h hn'',
+  rw [eval₂_sub, eval₂_X_pow, eval₂_C, sub_eq_zero_iff_eq] at hb,
+  have hb' : b ≠ 0,
+  { intro hb',
+    rw [hb', zero_pow hn'] at hb,
+    exact ha' hb.symm },
+  let s := ((X ^ n - C a).map i).roots,
+  have hs : _ = _ * (s.map _).prod := eq_prod_roots_of_splits h,
+  rw [leading_coeff_X_pow_sub_C hn', ring_hom.map_one, C_1, one_mul] at hs,
+  have hs' : s.card = n := (nat_degree_eq_card_roots h).symm.trans nat_degree_X_pow_sub_C,
+  apply @splits_of_exists_multiset F E _ _ i (X ^ n - 1) (s.map (λ c : E, c / b)),
+  rw [leading_coeff_X_pow_sub_one hn', ring_hom.map_one, C_1, one_mul, multiset.map_map],
+  have C_mul_C : (C (i a⁻¹)) * (C (i a)) = 1,
+  { rw [←C_mul, ←i.map_mul, inv_mul_cancel ha, i.map_one, C_1] },
+  have key1 : (X ^ n - 1).map i = C (i a⁻¹) * ((X ^ n - C a).map i).comp (C b * X),
+  { rw [map_sub, map_sub, map_pow, map_X, map_C, map_one, sub_comp, pow_comp, X_comp, C_comp,
+        mul_pow, ←C_pow, hb, mul_sub, ←mul_assoc, C_mul_C, one_mul] },
+  have key2 : (λ q : polynomial E, q.comp (C b * X)) ∘ (λ c : E, X - C c) =
+    (λ c : E, C b * (X - C (c / b))),
+  { ext1 c,
+    change (X - C c).comp (C b * X) = C b * (X - C (c / b)),
+    rw [sub_comp, X_comp, C_comp, mul_sub, ←C_mul, mul_div_cancel' c hb'] },
+  rw [key1, hs, prod_comp, multiset.map_map, key2, multiset.prod_map_mul, multiset.map_const,
+      multiset.prod_repeat, hs', ←C_pow, hb, ←mul_assoc, C_mul_C, one_mul],
+  all_goals { exact field.to_nontrivial F },
+end
+
+lemma gal_X_pow_sub_C_is_solvable (n : ℕ) (x : F) : is_solvable (X ^ n - C x).gal :=
+begin
+  by_cases hx : x = 0,
+  { rw [hx, C_0, sub_zero],
+    exact gal_X_pow_is_solvable n },
+  apply gal_is_solvable_tower (X ^ n - 1) (X ^ n - C x),
+  { exact splits_X_pow_sub_one_of_X_pow_sub_C _ n hx (splitting_field.splits _) },
+  { exact gal_X_pow_sub_one_is_solvable n },
+  { rw [map_sub, map_pow, map_X, map_C],
+    apply gal_X_pow_sub_C_is_solvable_aux,
+    have key := splitting_field.splits (X ^ n - 1 : polynomial F),
+    rwa [←splits_id_iff_splits, map_sub, map_pow, map_X, map_one] at key },
+end
+
+end gal_X_pow_sub_C
+
+variables (F)
+
+/-- Inductive definition of solvable by radicals -/
+inductive is_SBR : E → Prop
```

Comment:
Now that everything is stabilizing, what do you think of globally replacing `SBR` by `solvable_by_radicals`? Of maybe `solv_by_rad` if you think the former is too long. I think that the improved readability is worth it. And by using namespaces and dot-notation, I don't think the other statements will get that much longer.

Reply:
I just realized that I renamed it to `solvable_by_rad` rather than one of the two suggestions that you made. Is this fine or should I rename?

Reply:
I think it's fine